_posts/2023/2023-06-06-getting-started-with-vector-dbs-in-python.md

```diff
@@ -376,6 +376,151 @@ ENDPOINT=http://localhost:8080 CONFIG_ENDPOINT=http://localhost:19071 poetry run
 < DELETE http://localhost:19071/application/v2/tenant/default/application/default - 200
 {% endhighlight %}
 
+### Weaviate
+Weaviate is a vector search engine specifically designed for natural language and numerical data. It uses contextualized embeddings in data objects to understand semantic similarity. Currently, it supports only Hierarchical Navigable Small World(HNSW) indexing and is more costly on building data to indexes however has a fast query time and high scalability. Weaviate is open-source, easy to use, flexible, extensible and has a Contributor License Agreement.
+
+After you sign up at Weaviate Cloud Services WCS, and create a new free tier Weaviate Cluster with authentication. Note your cluster URL and API key (optional). The endpoint will have the following format https://myindex.weaviate.network.
+
+It is easy to create some objects with vectors.
+
+{% highlight python %}
+
+with Client(
+    event_hooks={
+        "request": [log_request],
+        "response": [log_response, Response.raise_for_status],
+    }
+) as client:
+    # index data
+    vectors = [
+        {
+            "id": "vec1",
+            "values": [0.1, 0.2, 0.3],
+            "properties": {
+                "genre": "drama"
+            }
+        },
+        {
+            "id": "vec2",
+            "values": [0.2, 0.3, 0.4],
+            "properties": {
+                "genre": "action"
+            }
+        }
+    ]
+
+    objects = []
+    for vector in vectors:
+        obj = {
+            "class": "Vectors",
+            "properties": {
+                "vector": vector["values"]
+            }
+        }
+        objects.append(obj)
+
+    client.post(urljoin(endpoint, "/v1/batch/objects"), json={"objects": objects}, headers=headers)
+{% endhighlight %}
+
+The search is pretty straight forward. Weaviate also has a GraphQl interface online.
```

Comment:
straightforward is one word.

GraphQL is double capital QL

remove "online".